### DISCUSSION TITLE
Error when providing an initial condition to an array variable

### DISCUSSION THREAD

**Discussion by nuclearkevin** [idaholab/moose#25306] [Q&A General] [answered]
Hello all,

I've been working on migrating some multi-group radiation transport code over to array variables and recently stumbled across a bit of an issue. MOOSE seems to throw a `_Map_base::at` error when an `ArrayMooseVariable` is initialized with `components = 1` and an initial condition is provided. This behaviour is not present when there are more components to the `ArrayVariable`, and using a dedicated `ConstantIC` doesn't get around the issue either. 

My main workaround at the moment is to use non-array variables for the case when `components = 1`, but this isn't a particularly elegant approach (requires kernels/bcs, etc. for both array and non-array variables) and I was wondering if there would be a better way to go about it. 

Here's a minimal example of the issue itself:
```
[Mesh]
  [Test]
    type = CartesianMeshGenerator
    dim = 2
    dx = '1.0'
    dy = '1.0'
    ix = '1'
    iy = '1'
  []
[]

[Variables]
  [Test]
    type = ArrayMooseVariable
    order = FIRST
    family = LAGRANGE
    components = 1
    initial_condition = '0.0'
  []
[]

[Problem]
  type = FEProblem
  solve = false
  kernel_coverage_check = false
  material_coverage_check = false
[]

[Executioner]
  type = Steady
[]

[Outputs]
  exodus = true
[]
```

Cheers,
Kevin

**Comment by lindsayad** [comment, marked as answer]
Have you tried using `ArrayConstantIC`?

**Reply by nuclearkevin** [reply]
I had not - just gave it a go and it solves the issue. Thank you for the fast response!
